File: Frontend/lib/logger.ts
export const logger = {
  info: (message: string, data?: any) => console.log(`[INFO] ${message}`, data || ''),
  error: (message: string, error?: any) => {
    console.error(`[ERROR] ${message}`, error || '');
    if (error?.response) {
      console.error('Response data:', error.response.data);
      console.error('Status:', error.response.status);
    }
  },
  debug: (message: string, data?: any) => console.debug(`[DEBUG] ${message}`, data || '')
};

File: Backend/app/utils/logger.py
"""Logging configuration."""

import logging
import sys


def setup_logging(debug: bool = False) -> None:
    """
    Configure application logging.
    
    Args:
        debug: Enable debug level logging
    """
    level = logging.DEBUG if debug else logging.INFO
    
    logging.basicConfig(
        level=level,
        format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
        handlers=[
            logging.StreamHandler(sys.stdout)
        ]
    )
    
    # Set httpx logging to WARNING to reduce noise
    logging.getLogger("httpx").setLevel(logging.WARNING)


def get_logger(name: str) -> logging.Logger:
    """
    Get logger instance.
    
    Args:
        name: Logger name (typically __name__)
        
    Returns:
        Configured logger
    """
    return logging.getLogger(name)
